### DISCUSSION TITLE
Few questions on indexers

### DISCUSSION THREAD

**Discussion by bbfred** [microsoft/AI-Chat-App-Hack#62] [Q&A]
Hi,

I have a few questions on indexer, I am looking into evaluating the retrieval in a multilingual set-up and thus wanted to test multiple embedding strategies. Is it possible to do the following on Search AI:

- Is it possible to store multiple vectors in a single index? For example, one vector for embedding in model A and an other one for model B etc. And how to switch the vector used for the search?

- Where would it be best to store information to then use as filter in the retrieval stage?

- Is there a way to index on two separate indexes, one for the chunks and one for the parent documents? Ideally, using one indexer to populate the two indexes at the same time and relate documents across indexes with an id.

Thanks,

Best,

Fred

**Comment by pamelafox** [comment]
Sorry I missed this question!

* Yes, you can store multiple vectors in a single index. Our repo (https://github.com/Azure-Samples/azure-search-openai-demo/) already does that for the case of GPT-4-vision support, it stores embeddings of both the text and image in two different fields. See slide 13 here: https://speakerdeck.com/pamelafox/azure-ai-search-best-practices-for-rag-chat-apps?slide=13
* You can store information for filtering in additional fields. You can see how we set up authentication to be able to search by oid and groups in https://github.com/Azure-Samples/azure-search-openai-demo/ as an example.

     ```
                fields.append(
                    SimpleField(
                        name="oids", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True
                    )
                )
     ```

* Are the parent documents the entire text of the document? i.e. chunks in one, whole text in the other? That seems doable, you'd just create the id yourself and set it on fields in both indexes. You might also look to see how the integrated vectorization support works to see if that already does that- https://github.com/Azure-Samples/azure-search-openai-demo/pull/1159/files

**Reply by srbalakr** [reply]
Regarding 

> Is there a way to index on two separate indexes, one for the chunks and one for the parent documents? Ideally, using one indexer to populate the two indexes at the same time and relate documents across indexes with an id.

It works opposite. Multiple indexers from different DataSource can point to single index.
